src/main/java/nl/hsac/fitnesse/fixture/util/XmlHttpResponse.java

```diff
@@ -22,7 +23,25 @@
     @Override
     public void validResponse() {
         super.validResponse();
+        checkXmlResponse();
+    }
+
+    @Override
+    public boolean responseIsValid(boolean throwExceptionOnHttpRequestFailure) {
+        boolean responseIsValid = super.responseIsValid(throwExceptionOnHttpRequestFailure);
+        if (throwExceptionOnHttpRequestFailure) {
+            checkXmlResponse();
+        } else {
+            try {
+                checkXmlResponse();
```

Comment:
No need to duplicate the call, you can always do the try/catch and in the catch block decide whether you want to re-throw the exception or just set responseIsValid to false. Or am I missing something?